mods/persistence/controllers/subsystems/mapping.dm

```diff
@@ -17,20 +17,46 @@
 		maploader.load_map(map_file, 1, 1, text2num(z), no_changeturf = TRUE)
 		CHECK_TICK
 	for(var/z in GLOB.using_map.station_levels)
-		if(z == 1)
+		// The highest z-level shouldn't have z_levels[z] set to true.
+		if(z == GLOB.using_map.station_levels[GLOB.using_map.station_levels.len])
 			continue
+		if (z_levels.len < z)
+			z_levels.len = z
 		z_levels[z] = TRUE
 
 	// Build the list of static persisted levels from our map.
 #ifdef UNIT_TEST
 	report_progress("Unit testing, so not loading saved map")
 #else
 	report_progress("Loading world save.")
+	
 	SSpersistence.LoadWorld()
+	
+	// Initialize the overmap for Persistence.
+	if(GLOB.using_map.use_overmap)
+		if(!GLOB.using_map.overmap_z)
+			build_overmap() // If a overmap hasn't been loaded, create a new one.
+
+	// Generate the areas of the overmap that are hazardous.
+	// Map is placed at the lower left corner of the overmap, not including the edges (2, 2).
+	new /datum/random_map/automata/overmap(null,2,2,GLOB.using_map.overmap_z,GLOB.using_map.overmap_size-2,GLOB.using_map.overmap_size-2,0,1,1)
```

Comment:
reeee magic numbers

why -2?

Reply:
The edges of the overmap are composed of a different tile that is used for looping. Because of this, the edges can't have the overmap automata overlaid on top of it, so the -2, -2 while applying the map at 2, 2 just ignores the edges.